posix/subsystem/src/inotify.cpp

```diff
@@ -29,14 +32,35 @@ struct OpenFile : File {
 		: file{file_}, descriptor{descriptor}, mask{mask} { }
 
 		void observeNotification(uint32_t events,
-				const std::string &name, uint32_t cookie) override {
+				const std::string &name, uint32_t cookie, bool isDir) override {
 			uint32_t inotifyEvents = 0;
 			if(events & FsObserver::deleteEvent)
 				inotifyEvents |= IN_DELETE;
+			if(events & FsObserver::deleteSelfEvent)
+				inotifyEvents |= IN_DELETE_SELF;
 			if(events & FsObserver::createEvent)
 				inotifyEvents |= IN_CREATE;
+			if(events & FsObserver::modifyEvent)
+				inotifyEvents |= IN_MODIFY;
+			if(events & FsObserver::accessEvent)
+				inotifyEvents |= IN_ACCESS;
+			if(events & FsObserver::closeWriteEvent)
+				inotifyEvents |= IN_CLOSE_WRITE;
+			if(events & FsObserver::closeNoWriteEvent)
+				inotifyEvents |= IN_CLOSE_NOWRITE;
+
+			auto preexists = std::ranges::find_if(file->_queue, [&name, inotifyEvents](auto a) {
```

Comment:
Just capture everything by reference here (IMO individual capture doesn't really make sense for short functions):
```suggestion
			auto preexists = std::ranges::find_if(file->_queue, [&](auto a) {
```